lib/test/mdx_test.ml

```diff
@@ -300,36 +349,30 @@ let run_exn ~non_deterministic ~silent_eval ~record_backtrace ~syntax ~silent
       | Include { file_included; file_kind = Fk_other _ } ->
           let new_content = read_part file_included None in
           update_block_content ?syntax ppf t new_content
-      | OCaml { non_det; env; errors } ->
+      | OCaml { non_det; env; errors; header = _ } ->
           let det () =
             assert (syntax <> Some Cram);
             Mdx_top.in_env env (fun () ->
-                eval_ocaml ~block:t ?syntax ?root c ppf t.contents errors)
+                eval_ocaml ~block:t ?syntax ?root c ppf errors)
           in
-          with_non_det non_deterministic non_det ~command:print_block
-            ~output:det ~det
+          with_non_det non_deterministic non_det ~on_skip_execution:print_block
+            ~on_keep_old_output:det ~on_evaluation:det
       | Cram { language = _; non_det } ->
-          let pad, tests =
-            Cram.of_lines
-              ~syntax:(Option.value ~default:Normal syntax)
-              ~loc:t.loc t.contents
-          in
-          with_non_det non_deterministic non_det ~command:print_block
-            ~output:(fun () ->
+          (* t.contents for md files is wrong, it's missing the begin and end *)
```

Comment:
I'm not quite clear about it myself anymore.

I think the issue is that in markdown `t.contents` is missing the initial newline (and potential trailing newline, if it doesn't end the file), so there's another place where the parsers for the various formats have slightly different behavior.

I think this is another hole to dig and can be tackled at some other point in time. The comment is quite unclear so to prevent people from wondering what this cryptic scribble means I'll remove it.